packages/mobile/src/pincode/authentication.ts

```diff
@@ -319,8 +381,12 @@ export async function updatePin(account: string, oldPin: string, newPin: string)
     if (updated) {
       clearPasswordCaches()
       setCachedPin(DEFAULT_CACHE_ACCOUNT, newPin)
-      const hash = await getPasswordHash(newPassword)
+      const hash = getPasswordHash(newPassword)
       await storePasswordHash(hash, account)
+      const pincodeType = pincodeTypeSelector(store.getState())
+      if (pincodeType === PincodeType.PhoneAuth) {
```

Comment:
I'm _a bit_ worried about what happens if an intermediate steps fail, but I'm not sure if the worry makes sense or not 🤔  We have a blanket error catch, but say we save the PIN this way but then updating the mnemonic fails. I guess that's where the asking for PIN fallback comes in 😄 

Yeah, writing this I think I convinced myself that it's fine, but sending the message anyways just in case

Reply:
yeah indeed, i've tried to make the phone auth as "optional" as possible - basically just try and if it fails then the user just gets the flow they currently have